### ISSUE TITLE
LinAlgError thrown when passing certain functions into relationship

### ISSUE THREAD

**Issue by ff995** [arm61/uravu#23]
Hello

Sorry to bother you again. But when working with certain functions and arrays, the relationship object throws a LinAlgError citing a singular matrix. Do you know which factors may cause this? It seems to occur during line 59 of code within Relationship: 

if ordinate_error is not None:
            potential_y.append(Distribution(stats.norm.rvs(loc=y, scale=ordinate_error[i], size=5000)))
I have attached instances where the object works and when it throws an error.

![image](https://user-images.githubusercontent.com/70442131/92257447-26824780-eecd-11ea-8c55-04fe6b8230c0.png)
![image](https://user-images.githubusercontent.com/70442131/92257514-4285e900-eecd-11ea-9d2f-c2b107f3fbe0.png)
![image](https://user-images.githubusercontent.com/70442131/92257607-6b0de300-eecd-11ea-89cc-30a33b8711a9.png)

**Comment by arm61** [comment]
This is a good point that I should add some documentation or a helpful error message about! So it is because the uncertainty (`dy`) in this example code has 0 values, so when the kernel density approximation is performed to get a description of the data likelihood form it sees a delta-function and returns a singular matrix. 

TLDR; your uncertainty has 0 values which `uravu` considers physically unrealistic. 

I am gonna leave the issue open so that I can add some documentation and an error message.

**Comment by arm61** [comment]
Thanks for using `uravu` though!

**Comment by ff995** [comment]
Thank you for responding. I have changed the value of dy to that used in the working example, but I still get the same error. I think in the 0*y case, dy is an array of zeroes.
![image](https://user-images.githubusercontent.com/70442131/92264339-01460700-eed6-11ea-913c-bf21c03aa608.png)
![image](https://user-images.githubusercontent.com/70442131/92264550-54b85500-eed6-11ea-9887-845bd07c3cb2.png)

**Comment by arm61** [comment]
You still have a 0 value in the `dy` value, because `x[0] = 0`, `y[0] = 4 * x[0] = 0`, and `dy[0] = y[0] * 0.2 = 0`. Does that make sense?

If you are interested in contributing a page for the documentation, or adding an error message related to this issue please feel free to open a pull request 😄

**Comment by ff995** [comment]
Ah yes it works fine now, my apologies for misunderstanding you and thanks for your help.

**Comment by arm61** [comment]
Awesome! Thanks for using `uravu`, if you have any other comments/issues please get in touch!

**Comment by arm61** [comment]
Added a nice error message in 450c6dc644687ebd6247fa27894a017b75984b53.
